impedance/models/circuits/fitting.py

```diff
@@ -148,6 +153,10 @@ def circuit_fit(frequencies, impedances, circuit, initial_guess, constants={},
         if weight_by_modulus:
             abs_Z = np.abs(Z)
             kwargs['sigma'] = np.hstack([abs_Z, abs_Z])
+        elif weight_proportionally and not weight_by_modulus:
```

Comment:
The extra check for weight_by_modulus isn't needed here since this line won't be reached if weight_by_modulus is True

```suggestion
        elif weight_proportionally:
```